scripts/data_validation/events_vs_expanded.py

```diff
@@ -0,0 +1,212 @@
+import os
+import json
+import logging
+import pandas as pd
+from google.cloud import bigquery
+
+
+def run_canvas_query(query_array, query_params):
+
+    # Instantiates a client
+    bigquery_client = bigquery.Client()
+
+    query_string = ''.join(query_array)
+    print(query_string)
+
+    # BQ Total Bytes Billed to report to status
+    total_bytes_billed = 0
+
+    job_config = bigquery.QueryJobConfig()
+    job_config.query_parameters = query_params
+
+    # Location must match that of the dataset(s) referenced in the query.
+    bq_job = bigquery_client.query(
+        query_string, location='US', job_config=job_config)
+    # This is the call that could result in an exception
+    df = bq_job.result().to_dataframe()
+
+    # print(df.to_string())
+    total_bytes_billed += bq_job.total_bytes_billed
+
+    return df
+
+
+def compare_expanded_vs_events_df(expanded_query_array, events_query_array, query_params):
+
+    expanded_df = run_canvas_query(
+        expanded_query_array, query_params)
+
+    events_df = run_canvas_query(
+        events_query_array, query_params)
+
+    # compare the two dataframes
+    # print(expanded_df.columns.values)
+    # print(events_df.columns.values)
+    print(f"shape of dataframe from expanded table: {expanded_df.shape}")
+    print(f"shape of dataframe from events table: {events_df.shape}")
+    pd.testing.assert_frame_equal(expanded_df, events_df)
+
+
+def main():
+    """
+    Get the configuration file
+    """
+    logger = logging.getLogger(__name__)
+
+    # Set up ENV
+    CONFIG_PATH = os.path.join(os.path.dirname(
+        os.path.abspath(__file__)), 'env.json')
+    try:
+        with open(CONFIG_PATH) as env_file:
+            ENV = json.load(env_file)
+    except FileNotFoundError:
+        logger.error(
+            f'Configuration file could not be found; please add file "{CONFIG_PATH}".')
+        ENV = dict()
+
+    DATA_WAREHOUSE_COURSE_IDS = ENV["DATA_WAREHOUSE_COURSE_IDS"]
+    DATA_WAREHOUSE_SHORT_COURSE_IDS = ENV["DATA_WAREHOUSE_SHORT_COURSE_IDS"]
+    CANVAS_DATA_ID_INCREMENT = ENV["CANVAS_DATA_ID_INCREMENT"]
+    TIME_LIMIT = ENV["TIME_LIMIT"]
+
+    query_params = [
+        bigquery.ArrayQueryParameter(
+            'course_ids', 'STRING', DATA_WAREHOUSE_COURSE_IDS),
+        bigquery.ArrayQueryParameter(
+            'course_ids_short', 'STRING', DATA_WAREHOUSE_SHORT_COURSE_IDS),
+        bigquery.ScalarQueryParameter(
+            'canvas_data_id_increment', 'INT64', CANVAS_DATA_ID_INCREMENT),
+        bigquery.ScalarQueryParameter(
+            'time_limit', 'STRING', TIME_LIMIT)
+    ]
+
+    # compare the Canvas query
+    canvas_expanded_query_array = [
```

Comment:
I think it's fine to have these hardcoded here. I wonder if in future validation queries we can have this include these from another file so it actually keeps them in sync with the code and doesn't repeat queries that are present somewhere else in the code.

Reply:
Or have these queries be able to be pulled on a config file, like they currently are. I think the ideal is that these were moved to the database though rather than being in files, since these are all Michigan specific queries.

Reply:
Moved query strings into the env_sample.json file, and simplified the events_vs_expanded.py file.